File: solutions/Virtual-Assistant/src/csharp/Microsoft.Bot.Solutions/Skills/SkillConfigurationBase.cs
using System.Collections.Generic;
using Microsoft.Bot.Builder.Azure;

namespace Microsoft.Bot.Solutions.Skills
{
    public abstract class SkillConfigurationBase
    {
        public bool IsAuthenticatedSkill { get; set; }

        public abstract Dictionary<string, string> AuthenticationConnections { get; set; }

        public abstract CosmosDbStorageOptions CosmosDbOptions { get; set; }

        public abstract Dictionary<string, LocaleConfiguration> LocaleConfigurations { get; set; }

        public abstract Dictionary<string, object> Properties { get; set; }
    }
}

File: solutions/Virtual-Assistant/src/typescript/bot-solution/src/skills/skillConfigurationBase.ts
import { CosmosDbStorageSettings } from 'botbuilder-azure';
import { LocaleConfiguration } from './localeConfiguration';

export abstract class SkillConfigurationBase {

    public isAuthenticatedSkill: boolean = false;

    public abstract authenticationConnections: { [key: string]: string };

    public abstract cosmosDbOptions: CosmosDbStorageSettings;

    public abstract localeConfigurations: Map<string, LocaleConfiguration>;

    public abstract properties: { [key: string]: Object|undefined };
}
